### PR TITLE
gateway-api: Add tests for standard CRD

### PR DESCRIPTION
**Pull request by sayboras**
This is to improve the coverage with the standard CRD channel. Also, small changes on existing docs are done.

**Note**: required GHA jobs should be updated after this PR is merged.

### CODE REVIEW

File: Documentation/network/servicemesh/gateway-api/installation.rst [hunk 1]
```diff
@@ -5,21 +5,24 @@ Prerequisites
...
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v0.5.1/config/crd/standard/gateway.networking.k8s.io_httproutes.yaml
-        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v0.5.1/config/crd/experimental/gateway.networking.k8s.io_referencegrants.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v0.7.0/config/crd/standard/gateway.networking.k8s.io_gatewayclasses.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v0.7.0/config/crd/standard/gateway.networking.k8s.io_gateways.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v0.7.0/config/crd/standard/gateway.networking.k8s.io_httproutes.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v0.7.0/config/crd/standard/gateway.networking.k8s.io_referencegrants.yaml
+        $ kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/v0.7.0/config/crd/standard/gateway.networking.k8s.io_referencegrants.yaml
```
Comment (mhofstetter) [inline Documentation/network/servicemesh/gateway-api/installation.rst]: looks like `referencegrants` is a duplicate.

File: .github/workflows/conformance-gateway-api.yaml [hunk 2]
```diff
@@ -105,11 +121,12 @@ jobs:
...
-          kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/${{ env.gateway_api_version }}/config/crd/experimental/gateway.networking.k8s.io_httproutes.yaml
+          kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/${{ env.gateway_api_version }}/config/crd/${{ matrix.crd-channel }}/gateway.networking.k8s.io_gatewayclasses.yaml
+          kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/${{ env.gateway_api_version }}/config/crd/${{ matrix.crd-channel }}/gateway.networking.k8s.io_gateways.yaml
+          kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/${{ env.gateway_api_version }}/config/crd/${{ matrix.crd-channel }}/gateway.networking.k8s.io_httproutes.yaml
+          kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/${{ env.gateway_api_version }}/config/crd/${{ matrix.crd-channel }}/gateway.networking.k8s.io_referencegrants.yaml
+          ## TLSRoute is only available in experimental channel in v0.7.0
```
Comment (mhofstetter) [inline .github/workflows/conformance-gateway-api.yaml:128]: shouldn't the experimental `TLSRoute` CRD only be applied in the corresponding matrix `experimental`? Or does the testrunner expect the CRD to be installed even though the feature isn't enabled?

Comment (sayboras) [reply .github/workflows/conformance-gateway-api.yaml:128]: we don't have granular control in operator, so without tlsroute crd, the operator watcher will fail and panic.

File: Documentation/network/servicemesh/gateway-api/installation.rst (hunk 1 above)
Comment (sayboras) [reply Documentation/network/servicemesh/gateway-api/installation.rst]: oops, good catch :+1:

File: .github/workflows/conformance-gateway-api.yaml (hunk 2 above)
Comment (mhofstetter) [reply .github/workflows/conformance-gateway-api.yaml:128]: ok, so it's expected that a user installs all the experimental CRDs too? :thinking: 
might be worth to add a feature toggle to the operator the same way its supported by the testrunner.
but definitely out of scope for this PR - thanks for the info!

**Review by mhofstetter — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]
Looks good, thanks!

**Review by nbusseneau — APPROVED** [review]

**Pull request merged**